### PR TITLE
Feature/post trip

### LINKED ISSUE #17: Traveler - Trip Request
I will select a date, duration, number of travelers and choose from a list of destinations

### LINKED ISSUE #18: Traveler - Trip Request - Cost
After making these selections, I should see an estimated cost (with a 10% travel agent fee) for the trip.

### LINKED ISSUE #19: Traveler - Trip Request - Displays on Dash
Once I submit the trip request, it will show on my dashboard as “pending” so that the travel agency can approve or deny it.

### PR DESCRIPTION
**Pull request by darlaevans2000**
# Description
Added post new trip method in API Calls. Added Dom updates for posting a new trip. Added booking form error handling (including no decimals). Added post of the new trip with no errors in scripts.js. 
closes #17 
closes #18 
closes #19 

## Type of change
- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
NPM Start and posting a new trip 

# Where should the reviewer start?:
apiCalls.js and scripts.js

### CODE REVIEW

File: dist/index.html [hunk 1]
```diff
@@ -77,19 +77,20 @@ <h2 for="bookingForm" class="book-top">
...
         <div class="selection">
         <label for="durationInput">Duration of trip:
-          <input required name="duration-input" id="durationInput" type="number" min="1" placeholder="number of nights">
+          <input required name="duration-input" id="durationInput" type="number" oninput="this.value=(parseInt(this.value)||0)" min="1" step='1' placeholder="number of nights">
         </label>
         </div>
         <div class="selection">
           <label for="travelersInput">Number of travelers:
-            <input required name="travelers-input" id="travelersInput" type="number" min="1">
+            <input required name="travelers-input" id="travelersInput" type="number" oninput="this.value=(parseInt(this.value)||0)" min="1" step='1' placeholder="number of travelers">
```
Comment (Eva-Navarrete89) [inline dist/index.html:85]: Nice format.

File: src/apiCalls.js [hunk 2]
```diff
@@ -30,8 +30,8 @@ const apiCalls = {
       })
   },
 
-    getSingleTraveler(userID) {
-    return fetch(`http://localhost:3001/api/v1/travelers/${userID}`)
+    getSingleTraveler() {
+    return fetch(`http://localhost:3001/api/v1/travelers/1`)
```
Comment (Eva-Navarrete89) [inline src/apiCalls.js:34]: Great use of parameter. Making it dynamic!

File: src/apiCalls.js [hunk 3]
```diff
@@ -40,13 +40,28 @@ const apiCalls = {
...
+      body: JSON.stringify(tripData), 
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    })
+      .then(response => {
+        if (!response.ok) {
+          throw new Error(`Sorry, we could not process your booking request.`);
+        }
+          return response.json()
```
Comment (Eva-Navarrete89) [inline src/apiCalls.js:55]: Nice looking fetch and POST.

File: src/scripts.js [hunk 4]
```diff
@@ -62,7 +75,7 @@ function getRandomDate() {
 
 function checkLogin(event) {
   event.preventDefault();
-  if (passwordInput.value !== 'travel2020' || !passwordInput.value || !username || username.length < 9) {
+  if (passwordInput.value !== 'travel2020' || !passwordInput.value || !usernameInput.value || usernameInput.length < 9) {
```
Comment (Eva-Navarrete89) [inline src/scripts.js:78]: Nice if statements

File: src/scripts.js [hunk 5]
```diff
@@ -93,35 +110,94 @@ function displayUserInfo(traveler) {
...
+        newTripInstance.estimateTotalTripCost();
+        domUpdates.buildBookingMessage(newTripInstance)
+        domUpdates.displayTrips(currentTraveler.allTrips, cardGrid, "My Trips", allDestinations)
+        bookingForm.reset();
+        apiCalls.getAllData()
+        retrieveAllData()
+        // console.log(currentTraveler.allTrips)
+        // domUpdates.displayAnnualCosts(currentTraveler.annualCost);
+        // console.log(currentTraveler.annualCost)
+  }
```
Comment (Eva-Navarrete89) [inline src/scripts.js:202]: This looks great!

**Comment by darlaevans2000** [comment]
@hannahhch Second PR

**Pull request merged**